### PR TITLE
simplify ParametersAcceptorSelectorVariantsWrapper

### PR DESCRIPTION
**Pull request by staabm**

### CODE REVIEW

File: packages/NodeTypeResolver/PHPStan/ParametersAcceptorSelectorVariantsWrapper.php [hunk 1]
```diff
@@ -20,16 +20,11 @@ public static function select(
...
 
-        if ($node->isFirstClassCallable()) {
+        if ($node instanceof CallLike && $node->isFirstClassCallable()) {
             return ParametersAcceptorSelector::selectSingle($variants);
         }
 
-        return count($variants) > 1
-            ? ParametersAcceptorSelector::selectFromArgs($scope, $node->getArgs(), $variants)
-            : ParametersAcceptorSelector::selectSingle($variants);
+        return ParametersAcceptorSelector::selectFromArgs($scope, $node->getArgs(), $variants);
```
Comment (samsonasik) [inline packages/NodeTypeResolver/PHPStan/ParametersAcceptorSelectorVariantsWrapper.php:32]: there is no `getArgs()` method in `FunctionLike`

Comment (staabm) [reply packages/NodeTypeResolver/PHPStan/ParametersAcceptorSelectorVariantsWrapper.php:32]: good catch, thx

File: packages/NodeTypeResolver/PHPStan/ParametersAcceptorSelectorVariantsWrapper.php [hunk 2]
```diff
@@ -20,16 +20,15 @@ public static function select(
...
-
+        
         if ($node->isFirstClassCallable()) {
             return ParametersAcceptorSelector::selectSingle($variants);
         }
 
-        return count($variants) > 1
-            ? ParametersAcceptorSelector::selectFromArgs($scope, $node->getArgs(), $variants)
-            : ParametersAcceptorSelector::selectSingle($variants);
+        return ParametersAcceptorSelector::selectFromArgs($scope, $node->getArgs(), $variants);
```
Comment (samsonasik) [inline packages/NodeTypeResolver/PHPStan/ParametersAcceptorSelectorVariantsWrapper.php:32]: `selectFromArgs()` is slower than `selectSingle()` as far as I remember, as it check many things, verify `count() > 1` early was already faster.

Comment (staabm) [reply packages/NodeTypeResolver/PHPStan/ParametersAcceptorSelectorVariantsWrapper.php:32]: at least in phpstan itself we try to get rid of `selectSingle` occurences and use `selectFromArgs` where possible.

I did not yet see it in perf profiles, but you are likely right, that `selectSingle` is marginally faster

Comment (samsonasik) [reply packages/NodeTypeResolver/PHPStan/ParametersAcceptorSelectorVariantsWrapper.php:32]: `count()` without recursive call is cheap, while call `selectFromArgs()` that check many things already will be slower, so I prefer the original implementation to check `count()` first, then use `selectSingle()` when possible, fallback to `selectFromArgs()` on multi variants.

Comment (staabm) [reply packages/NodeTypeResolver/PHPStan/ParametersAcceptorSelectorVariantsWrapper.php:32]: 👍

**Pull request closed without merging**